sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -45,7 +46,12 @@ trait CodegenSupport extends SparkPlan {
     case _: SortMergeJoinExec => "smj"
     case _: RDDScanExec => "rdd"
     case _: DataSourceScanExec => "scan"
-    case _ => nodeName.toLowerCase(Locale.ROOT)
```

Comment:
This caused one of the tests to fail as the nodename generated was not a single word (like before) but something like `scan in-memory my_table`... which does not compile with codegen. The change done was to retain only the alpha-numeric characters in the `nodeName`